packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -753,6 +773,23 @@ export class LitDevTutorial extends LitElement {
       projectSrcAfter: `${this._samplesRoot}/tutorials/${this._projectLocation}/${afterSlug}/project.json`,
     };
   }
+
+  private reportMetrics(idx: number, tutorialUrl: string) {
+    if (idx === 0 || !hasRecordedStart) {
+      window.gtag?.('event', 'tutorial_start', {
+        category: 'tutorials',
+        event_label: tutorialUrl,
+        value: idx,
+      });
+      hasRecordedStart = true;
+    } else if (idx === this._manifest.steps.length - 1 && !hasRecordedEnd) {
+      window.gtag?.('event', 'tutorial_end', {
+        category: 'tutorials',
+        event_label: tutorialUrl,
+      });
+      hasRecordedEnd = true;
+    }
+  }
```

Comment:
Is it worth splitting out the gtag related logic into a separate analytics file, and making this a pure function that isn't bound to the component?